src/main/java/com/blackducksoftware/integration/hub/alert/web/controller/ConfigController.java

```diff
@@ -22,92 +22,79 @@
-            if (response == null) {
-                final D updatedEntity = repository.save(modelEntity);
-                response = createResponse(HttpStatus.ACCEPTED, updatedEntity.getId(), "Updated.");
+    public ResponseEntity<String> putConfig(final R restModel) throws IntegrationException {
+        if (configActions.doesConfigExist(restModel.getId())) {
+            final Map<String, String> validationResults = configActions.validateConfig(restModel);
+            if (validationResults.isEmpty()) {
+                configActions.customTriggers(restModel);
+                final D updatedEntity = configActions.saveConfig(restModel);
+                return createResponse(HttpStatus.CREATED, updatedEntity.getId(), "Updated");
+            } else {
+                final ResponseBodyBuilder responseBuilder = new ResponseBodyBuilder(configActions.objectTransformer.stringToLong(restModel.getId()), "There were errors with the configuration.");
+                responseBuilder.putErrors(validationResults);
+                return new ResponseEntity<>(responseBuilder.build(), HttpStatus.BAD_REQUEST);
             }
-            return response;
         }
         return createResponse(HttpStatus.BAD_REQUEST, restModel.getId(), "No configuration with the specified id.");
     }
 
     public abstract ResponseEntity<String> validateConfig(R restModel);
 
     public ResponseEntity<String> deleteConfig(final R restModel) {
-        if (restModel.getId() != null && repository.exists(restModel.getId())) {
-            repository.delete(restModel.getId());
-            return createResponse(HttpStatus.ACCEPTED, restModel.getId(), "Deleted.");
+        if (configActions.doesConfigExist(restModel.getId())) {
+            configActions.deleteConfig(restModel.getId());
+            return createResponse(HttpStatus.ACCEPTED, restModel.getId(), "Deleted");
         }
         return createResponse(HttpStatus.BAD_REQUEST, restModel.getId(), "No configuration with the specified id.");
     }
 
-    public abstract ResponseEntity<String> testConfig(final R restModel);
+    public abstract ResponseEntity<String> testConfig(final R restModel) throws IntegrationException;
 
-    public abstract D restModelToDatabaseModel(final R restModel);
-
-    public abstract R databaseModelToRestModel(final D databaseModel);
-
-    public List<R> databaseModelsToRestModels(final List<D> databaseModels) {
-        final List<R> restModels = new ArrayList<>();
-        for (final D databaseModel : databaseModels) {
-            restModels.add(databaseModelToRestModel(databaseModel));
-        }
-        return restModels;
+    protected ResponseEntity<String> createResponse(final HttpStatus status, final String id, final String message) {
+        return createResponse(status, configActions.objectTransformer.stringToLong(id), message);
     }
 
     protected ResponseEntity<String> createResponse(final HttpStatus status, final Long id, final String message) {
```

Comment:
One of these should be private; based on your changes, I believe it should be this one.